ext/opcache/Optimizer/block_pass.c

```diff
@@ -111,7 +111,9 @@ static int get_const_switch_target(zend_cfg *cfg, zend_op_array *op_array, zend_
 	HashTable *jumptable = Z_ARRVAL(ZEND_OP2_LITERAL(opline));
 	zval *zv;
 	if ((opline->opcode == ZEND_SWITCH_LONG && Z_TYPE_P(val) != IS_LONG)
-			|| (opline->opcode == ZEND_SWITCH_STRING && Z_TYPE_P(val) != IS_STRING)) {
+			|| (opline->opcode == ZEND_SWITCH_STRING && Z_TYPE_P(val) != IS_STRING)
+			|| (opline->opcode == ZEND_MATCH_LONG && Z_TYPE_P(val) != IS_LONG)
```

Comment:
You were right - this still needs to be changed because the "next block" isn't the IS_EQUAL or CASE opcodes. When opcache runs, the following example goes to the wrong block

```
function test (bool $flag) {
    $x = match(0.0) {
        1, 2, 3, 4, 5 => 'some default',
    };
    var_export($x);
}
test(false);
```

Reply:
It looks as if `block->successors[block->successors_count - 2];` is always the `default =>` node (implicit or explicit).
So it should be that for `ZEND_MATCH_LONG` and `ZEND_MATCH_STRING` and continue to be `- 1` for `ZEND_SWITCH_*`
 
```
	if (!zv) {
		/* default */
		return block->successors[block->successors_count - 2];
	}
```

Reply:
Thanks! I've tested every variation I could come up with. Should be fixed now.